provider/pkg/provider/provider.go

```diff
@@ -1709,31 +1733,90 @@ func (k *azureNativeProvider) getAuthConfig() (*authentication.Config, error) {
 	return builder.Build()
 }
 
-func (k *azureNativeProvider) getAuthorizers(authConfig *authentication.Config) (tokenAuth autorest.Authorizer,
+func assertAzVersion() error {
+	const versionHint = "Please make sure the Azure CLI 2.37 or greater, but less than 3.x, is installed or configure another authentication method."
+
+	_, err := exec.LookPath("az")
+	if err != nil {
+		return fmt.Errorf("could not find `az`. %s", versionHint)
+	}
+
+	var azVersion struct {
+		Cli string `json:"azure-cli"`
+	}
+	err = jsonUnmarshalAzCmd(&azVersion, "version")
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("could not determine az version. %s", versionHint))
+	}
+
+	actual, err := goversion.NewVersion(azVersion.Cli)
+	if err != nil {
+		return fmt.Errorf("could not parse az version %q: %+v. %s", azVersion.Cli, err, versionHint)
+	}
+
+	if actual.LessThan(minAzVersion) || actual.GreaterThanOrEqual(nextMajorAzVersion) {
+		return fmt.Errorf("found incompatible az version %s. %s", actual, versionHint)
+	}
+
+	return nil
+}
+
+func jsonUnmarshalAzCmd(i interface{}, arg ...string) error {
+	var stderr bytes.Buffer
+	var stdout bytes.Buffer
+
+	cmd := exec.Command("az", arg...)
+
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+
+	if err := cmd.Run(); err != nil {
+		err := fmt.Errorf("running az: %+v", err)
```

Comment:
```suggestion
		err := fmt.Errorf("running az: %w", err)
```
Were you trying to dump the stack there? In general better to wrap the error instead I find and let the caller decide what to do with the error.